src/service/xhr-impl.js

```diff
@@ -576,6 +593,13 @@ export class FetchResponseHeaders {
  * @return {!Xhr}
  */
 export function xhrServiceForTesting(window) {
+  registerServiceBuilder(window, 'ampdoc', function() {
+    return {
+      isSingleDoc() {
+        return false;
+      },
+    };
+  });
```

Comment:
The hidden side effect would cause subsequent ampdoc registrations to silently fail. Perhaps we can either allow injection of the ampdoc instance or just don't call that code in test mode. 

The rest LGTM.